src/afrolabs/components/kafka.clj

```diff
@@ -1546,12 +1574,20 @@
                             (.get)
                             (set))
 
+        desired-topics (->> (mapcat #(get-topic-names %) topic-name-providers)
+                            (distinct))
```

Comment:
You could make `desired-topics` a set straight away. Most of the time you use it as a set, and the rest it is passed to a filter, which also works with sets.

Reply:
Made it a set and changed all of the seq operations to be native `set/intersection` instead of `(filter one-set second-set)`